src/System.Runtime.Extensions/tests/System/IO/PathTests.netcoreapp.cs

```diff
@@ -172,5 +172,177 @@ public static void GetFileNameWithSpaces_Unix_Span()
+			yield return new object[] { Path.Combine(Path.GetPathRoot(curDir), "somedir", ".."), curDir, Path.GetPathRoot(curDir) };
+			yield return new object[] { Path.Combine(Path.GetPathRoot(curDir), "."), curDir, Path.GetPathRoot(curDir) };
+			yield return new object[] { Path.Combine(Path.GetPathRoot(curDir), "..", "..", "..", ".."), curDir, Path.GetPathRoot(curDir) };
+			yield return new object[] { Path.GetPathRoot(curDir) + new string(Path.DirectorySeparatorChar, 3), curDir, Path.GetPathRoot(curDir) };
+
+			// Device paths
+			yield return new object[] { "foo", @"\\?\C:\ ", @"\\?\C:\ \foo" };
+			yield return new object[] { "foo", @"\\?\C:\ \ ", @"\\?\C:\ \ \foo" };
+			yield return new object[] { "foo", @"\\?\C:\ .", @"\\?\C:\ .\foo" };
+			yield return new object[] { "foo", @"\\?\C:\ ..", @"\\?\C:\ ..\foo" };
+			yield return new object[] { "foo", @"\\?\C:\...", @"\\?\C:\...\foo" };
+			yield return new object[] { "foo", @"\\?\GLOBALROOT\", @"\?\GLOBALROOT\foo" };
+			yield return new object[] { "foo", @"\\?\", @"\?\foo" };
+			yield return new object[] { "foo", @"\\?\.", @"\?\foo" };
+			yield return new object[] { "foo", @"\\?\..", @"\foo" };
+			yield return new object[] { "foo", @"\\?\\", @"\?\foo" };
+			yield return new object[] { "foo", @"\\?\C:\\", @"\\?\C:\\foo" };
+			yield return new object[] { "foo", @"\\?\C:\|", @"\\?\C:\|\foo" };
+			yield return new object[] { "foo", @"\\?\C:\.", @"\\?\C:\.\foo" };
+			yield return new object[] { "foo", @"\\?\C:\..", @"\\?\C:\..\foo" };
+			yield return new object[] { "foo", @"\\?\C:\Foo1\.", @"\\?\C:\Foo1\foo" };
+			yield return new object[] { "foo", @"\\?\C:\Foo2\..", @"\\?\C:\foo" };
+			yield return new object[] { "foo", @"\\?\UNC\", @"\\?\UNC\foo" };
+			yield return new object[] { "foo", @"\\?\UNC\server1", @"\\?\UNC\server1\foo" };
+			yield return new object[] { "foo", @"\\?\UNC\server2\", @"\\?\UNC\server2\foo" };
+			yield return new object[] { "foo", @"\\?\UNC\server3\\", @"\\?\UNC\server3\\foo" };
+			yield return new object[] { "foo", @"\\?\UNC\server4\..", @"\\?\UNC\server4\..\foo" };
+			yield return new object[] { "foo", @"\\?\UNC\server5\share\.", @"\\?\UNC\server5\share\foo" };
+			yield return new object[] { "foo", @"\\?\UNC\server6\share\..", @"\\?\UNC\server6\foo" };
+			yield return new object[] { "foo", @"\\?\UNC\a\b\\", @"\\?\UNC\a\b\foo" };
+			yield return new object[] { "foo", @"\\.\", @"\\.\foo" };
+			yield return new object[] { "foo", @"\\.\.", @"\\.\.\foo" };
+			yield return new object[] { "foo", @"\\.\..", @"\\.\..\foo" };
+			yield return new object[] { "foo", @"\\.\\", @"\\.\\foo" };
+			yield return new object[] { "foo", @"\\.\C:\\", @"\\.\C:\foo" };
+			yield return new object[] { "foo", @"\\.\C:\|", @"\\.\C:\|\foo" };
+			yield return new object[] { "foo", @"\\.\C:\.", @"\\.\C:\foo" };
+			yield return new object[] { "foo", @"\\.\C:\..", @"\\.\foo" };
```

Comment:
This is a bug- the root of `\\.\C:\.` should be `\\.\C:\`, just like `\\?\`. It would probably be good to break out the device paths into separate tests. Everything you did with normal paths should give you identical results with `\\?\` or `\\.\` in the front.

Relative paths are special, of course. I'd suggest:

GetFullPath_Windows_NotFullyQualified (no device paths)
GetFullPath_Windows_Unc (try with input data as is, then change to `\\?\` and `\\.\`)
GetFullPath_Windows_FullyQualified (try with input data as is, then change to `\\?\` and `\\.\`)

As there are paths in device syntax that can't be represented without the device syntax you'll also want to try changing the volume name to something other than `C:`.  E.g. `\\?\SomeVolume\`. Can probably just do so in the FullyQualified with the input data. (After just prepending, change the root completely)